plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -225,6 +231,57 @@ private DeltaLakeColumnHandle buildColumnHandle(EntryType entryType, CheckpointS
         return new DeltaLakeColumnHandle(entryType.getColumnName(), type, OptionalInt.empty(), entryType.getColumnName(), type, REGULAR, Optional.empty());
     }
 
+    /**
+     * Constructs a TupleDomain which filters on a specific required primitive sub-column of the EntryType being
+     * not null for effectively pushing down the predicate to the Parquet reader.
+     */
+    private TupleDomain<HiveColumnHandle> buildTupleDomainColumnHandle(EntryType entryType, HiveColumnHandle column)
+    {
+        String field;
+        Type type;
+        switch (entryType) {
+            case TRANSACTION:
+                field = "version";
+                type = BigintType.BIGINT;
+                break;
+            case ADD:
+                field = "path";
+                type = VarcharType.VARCHAR;
+                break;
+            case REMOVE:
+                field = "path";
+                type = VarcharType.VARCHAR;
+                break;
+            case METADATA:
+                field = "id";
+                type = VarcharType.VARCHAR;
+                break;
+            case PROTOCOL:
+                field = "minReaderVersion";
+                type = BigintType.BIGINT;
+                break;
+            case COMMIT:
+                field = "version";
+                type = BigintType.BIGINT;
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported Delta Lake checkpoint entry type: " + entryType);
+        }
+        HiveColumnHandle handle = new HiveColumnHandle(
+                column.getBaseColumnName(),
+                column.getBaseHiveColumnIndex(),
+                column.getBaseHiveType(),
+                column.getBaseType(),
+                Optional.of(new HiveColumnProjectionInfo(
+                        ImmutableList.of(-1), // hiveColumnIndex; we provide fake value because we always find columns by name
```

Comment:
nit: Use `0` - this is what i see in `io.trino.plugin.deltalake.DeltaLakeColumnHandle#toHiveColumnHandle`